### PR TITLE
Add documentation to pandas.core.groupby.GroupBy.apply

### PR DESCRIPTION
**Pull request by Rafalonso**
It took me a few hours to figure out how to edit rows from `groupBy` objects until I scavenged the code.

### CODE REVIEW

File: pandas/core/groupby.py [hunk 1]
```diff
@@ -162,6 +162,17 @@
     a    5
     b    2
     dtype: int64
+
+    When GroupBy object has group_keys=False changes are made to
```
Comment (TomAugspurger) [inline pandas/core/groupby.py]: I'm not sure what this means. Perhaps you could show the same example with and without `group_keys=False`? Or show an example where changing `group_keys` affects the output? It seems to be identical for both.

**Comment by jreback** [comment]
@pandas-dev/pandas-core needs a rebase

File: pandas/core/groupby.py (hunk 1 above)
Comment (gfyoung) [reply pandas/core/groupby.py]: @TomAugspurger : Indeed, the difference is not as evident as this new doc seems to imply:

https://github.com/pandas-dev/pandas/blob/486bfe8267cd8b71fb322f3b02db3921b0a4e14e/pandas/core/groupby/groupby.py#L1124-L1145

I'm uncertain as to whether this addition really clarifies anything...

**Comment by gfyoung** [comment]
@jreback : Given @TomAugspurger comment [above](https://github.com/pandas-dev/pandas/pull/19927#discussion_r171063351), I wonder if it might be best to close.  This PR has been dormant for a long time, and I'm not seeing the immediate benefit of this addition unfortunately.

**Pull request closed without merging**